test/test_fusion_profiler.cpp

```diff
@@ -21,10 +21,59 @@
 
 namespace nvfuser {
 
+namespace {
+
+class ProfilingOptsManager {
```

Comment:
Added this helper RAII class to help managing profiling options - without this the dirty state leaked to other tests and caused unexpected failures.

Reply:
@kevinstephano @wujingyue @jjsjann123 to review this code

Reply:
Hi Andrzej, I understand your motivation. However, are we at the risk of reinventing OptionsGuard? https://github.com/NVIDIA/Fuser/blob/51d087eddf18665d319bed1bd515eca1edadaf78/csrc/options.h#L172